src/main/java/com/browserstack/automate/ci/common/BrowserStackBuildWrapperOperations.java

```diff
@@ -144,8 +144,11 @@ public void buildEnvVars(Map<String, String> env) {
         }
 
         String isLocalEnabled = localConfig != null ? "true" : "false";
-        env.put(BrowserStackEnvVars.BROWSERSTACK_LOCAL, "" + isLocalEnabled);
-        logEnvVar(BrowserStackEnvVars.BROWSERSTACK_LOCAL, isLocalEnabled);
+
+        if (isLocalEnabled.equals("true")) {
+            env.put(BrowserStackEnvVars.BROWSERSTACK_LOCAL, "" + isLocalEnabled);
+            logEnvVar(BrowserStackEnvVars.BROWSERSTACK_LOCAL, isLocalEnabled);
+        }
 
         String localIdentifier =
```

Comment:
Handling of localIdentifier too? Don't set it explicit

Reply:
if no local config, it would be empty. Already there is a check